application/cms/page_service.py

```diff
@@ -145,6 +149,35 @@ def update_page(self, page, data, last_updated_by):
             self.logger.info(message)
             return page
 
+    def _set_data_sources(self, page, data_source_forms):
+        current_data_sources = page.data_sources
+        page.data_sources = []
+
+        for i, data_source_form in enumerate(data_source_forms):
+            existing_source = len(current_data_sources) > i
+
+            if get_bool(data_source_form.remove_data_source.data):
+                if existing_source:
+                    db.session.delete(current_data_sources[i])
+
+            else:
+                data_source = current_data_sources[i] if existing_source else DataSource()
+                data_source_form.populate_obj(data_source)
+
+                source_has_truthy_values = any(
```

Comment:
I don't get how this check has anything to do with truthy values?  Does this just mean any values at all?  i.e. we only add it to `page.data_sources` if something has been filled in?

Reply:
Yes, this is just a quick check that should validate we only add a data source if at least one field has some valid (i.e. "Truth"y) data.